docs/csharp/programming-guide/generics/constraints-on-type-parameters.md

```diff
@@ -10,72 +10,111 @@ helpviewer_keywords:
-  
-## Constraining Multiple Parameters  
- You can apply constraints to multiple parameters, and multiple constraints to a single parameter, as shown in the following example:  
-  
- [!code-csharp[csProgGuideGenerics#64](../../../csharp/programming-guide/generics/codesnippet/CSharp/constraints-on-type-parameters_4.cs)]  
-  
-## Unbounded Type Parameters  
- Type parameters that have no constraints, such as T in public class `SampleClass<T>{}`, are called unbounded type parameters. Unbounded type parameters have the following rules:  
-  
--   The `!=` and `==` operators cannot be used because there is no guarantee that the concrete type argument will support these operators.  
-  
--   They can be converted to and from `System.Object` or explicitly converted to any interface type.  
-  
--   You can compare to [null](../../../csharp/language-reference/keywords/null.md). If an unbounded parameter is compared to `null`, the comparison will always return false if the type argument is a value type.  
-  
-## Type Parameters as Constraints  
- The use of a generic type parameter as a constraint is useful when a member function with its own type parameter has to constrain that parameter to the type parameter of the containing type, as shown in the following example:  
-  
- [!code-csharp[csProgGuideGenerics#14](../../../csharp/programming-guide/generics/codesnippet/CSharp/constraints-on-type-parameters_5.cs)]  
-  
- In the previous example, `T` is a type constraint in the context of the `Add` method, and an unbounded type parameter in the context of the `List` class.  
-  
- Type parameters can also be used as constraints in generic class definitions. Note that the type parameter must be declared within the angle brackets together with any other type parameters:  
-  
- [!code-csharp[csProgGuideGenerics#15](../../../csharp/programming-guide/generics/codesnippet/CSharp/constraints-on-type-parameters_6.cs)]  
-  
- The usefulness of type parameters as constraints with generic classes is very limited because the compiler can assume nothing about the type parameter except that it derives from `System.Object`. Use type parameters as constraints on generic classes in scenarios in which you want to enforce an inheritance relationship between two type parameters.  
-  
-## See Also  
- <xref:System.Collections.Generic>  
+
+When you define a generic class, you can apply restrictions to the kinds of types that client code can use for type arguments when it instantiates your class. If client code tries to instantiate your class by using a type that is not allowed by a constraint, the result is a compile-time error. These restrictions are called constraints. Constraints are specified by using the `where` contextual keyword. The following table lists the six types of constraints:
```

Comment:
Again, we talk about *generic class* in the first sentence, though this topic also applies to generic methods and delegates?

The last sentence: six -> seven